core/services/multi.go

```diff
@@ -56,11 +56,23 @@ func (m *MultiStart) CloseBecause(reason error) (err error) {
 	return multierr.Append(reason, m.Close())
 }
 
-// MultiClose is a utility for closing multiple services concurrently.
-type MultiClose []io.Closer
+// CloseAll closes all elements concurrently.
+// Use this when you have various different types of io.Closer.
+func CloseAll(cs ...io.Closer) error {
```

Comment:
I experimented with returning an `io.Closer` since that seemed more consistent and composable, but it just created extra verbosity in the existing use cases.